File: app/src/main/java/com/example/dagger_hilt/ui/activity/splash/SplashViewModel.java
package com.example.dagger_hilt.ui.activity.splash;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.dagger_hilt.domain.ApiRepository;
import com.example.dagger_hilt.sys.util.Constants;

import java.util.concurrent.Executors;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class SplashViewModel extends ViewModel {
    ApiRepository apiRepository;

    @Inject
    public SplashViewModel(ApiRepository apiRepository) {
        this.apiRepository = apiRepository;
    }

    public MutableLiveData<Constants.StatusResponse> loadMovies() {
        var result = new MutableLiveData<Constants.StatusResponse>();
        var executor = Executors.newSingleThreadExecutor();
        executor.execute(() ->
                apiRepository.loadMovies(errorResource ->
                        result.postValue(errorResource.getStatus())));
        return result;
    }
}

File: app/src/main/java/com/example/dagger_hilt/ui/activity/splash/SplashViewModel.kt
package com.example.dagger_hilt.ui.activity.splash

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.dagger_hilt.domain.ApiRepository
import com.example.dagger_hilt.sys.util.Constants
import com.example.dagger_hilt.sys.util.Resource
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import javax.inject.Inject

@HiltViewModel
class SplashViewModel @Inject constructor(private val apiRepository: ApiRepository) :
    ViewModel() {
    fun loadMovies(): MutableLiveData<Constants.StatusResponse> {
        val result = MutableLiveData<Constants.StatusResponse>()
        viewModelScope.launch {
            withContext(Dispatchers.IO) {
                apiRepository.loadMovies { errorResource: Resource<String> ->
                    result.postValue(errorResource.status)
                }
            }
        }
        return result
    }
}